src/github.com/getlantern/idletiming/idletiming_listener.go

```diff
@@ -33,9 +32,13 @@ type idleTimingListener struct {
 func (l *idleTimingListener) Accept() (c net.Conn, err error) {
 	c, err = l.orig.Accept()
 	if err == nil {
-		c = Conn(c, l.idleTimeout, func() {
-			l.onIdle(c)
-		})
+		var onIdle func()
+		if l.onIdle != nil {
+			onIdle = func() {
+				l.onIdle(c)
+			}
```

Comment:
I would probably add a safeguard here in case `l.onIdle == nil` @oxtoacart, as in:

```
else {
   onIdle = func() {}
}
```

Otherwise if anything inside `Conn` fails to check for nil it'll panic!